### ISSUE TITLE
i want to make video 2x speed using moviepy but its making videos zoom in,enlarge

### ISSUE THREAD

**Issue by eswar159** [Zulko/moviepy#1309]
Hi 
I'm new to moviepy my requirement is to make a normal video into 2x. All the videos are recorded on the phone and the videos which are not required any rotations are working fine. By the videos which are needed to be rotated even, I apply rotation or even just trying to write the output is zoomed in with full screen here is the code

```

from moviepy.editor import VideoFileClip

from moviepy.audio import *

import moviepy.video.fx.all as vfx

clip = VideoFileClip("testingggg.mp4",audio=False) 

clip.size

#clip = clip.rotate(90) 

print("Duration of video : ", clip.duration)

print("Duration of video : ", clip.reader.fps)

clip = clip.speedx(2)

#clip = VideoFileClip("final.mp4",audio=False) 

#clip.size


clip.write_videofile("final.mp4", threads=4, audio_fps=44100,codec = 'libx264')


```

hope someone can help me here are some images so that u can get an idea on my issue image of the original file output is in this way 

![2020-09-10 (5)](https://user-images.githubusercontent.com/35969578/92725652-0c8fab80-f38a-11ea-836b-bd39658d0d22.png)
![2020-09-10 (6)](https://user-images.githubusercontent.com/35969578/92725659-0ef20580-f38a-11ea-8576-28082027df8b.png)

Thank you

**Comment by shih6** [comment]
have you solved you problem?
i have same problem

**Comment by shih6** [comment]
When I imported my video  The video is stretched

**Comment by eswar159** [comment]
> When I imported my video The video is stretched

yes and that happens only for videos which are not oriented properly(i mean which need rotation)
did u solve it
i was searching from past week but found none to resolve it

**Comment by shih6** [comment]
i just found it yet

**Comment by tburrows13** [comment]
You could manually rotate the videos 90 degrees using some other program, then import them into moviepy and rotate them back using clip.rotate().

**Comment by eswar159** [comment]
like my requirement is to 2x the video no need of rotation or any
but this is making video stretch in the above case 
how can i force not to stretch the video?

**Comment by shih6** [comment]
I have a lot of video to deal with and it doesn't have to appear

Do you have any software for batch processing

**Comment by shih6** [comment]
you can try to make a img background

**Comment by shih6** [comment]
clip1=mp.VideoFileClip(path)
        background = (mp.ColorClip((1280, 720), (0, 0, 0))
            .set_duration(clip1.duration)
            )
        final=mp.CompositeVideoClip([background,clip1])

**Comment by tburrows13** [comment]
> like my requirement is to 2x the video no need of rotation or any
> but this is making video stretch in the above case
> how can i force not to stretch the video?

It is likely a bug in moviepy. Rotated videos are an absolute pain to work with (from experience!).

**Comment by eswar159** [comment]
Thanks for the info @tburrows13 

> > like my requirement is to 2x the video no need of rotation or any
> > but this is making video stretch in the above case
> > how can i force not to stretch the video?
> 
> It is likely a bug in moviepy. Rotated videos are an absolute pain to work with (from experience!).

Thanks for the info @tburrows13

**Comment by eswar159** [comment]
> ```
>     clip1=mp.VideoFileClip(path)
>     background = (mp.ColorClip((1280, 720), (0, 0, 0))
>         .set_duration(clip1.duration)
>         )
>     final=mp.CompositeVideoClip([background,clip1])
> ```

I will try it now!

**Comment by shih6** [comment]
It's too much trouble to use ffmpeg
but i dont konw there is same troule in moviepy

**Comment by eswar159** [comment]
> ```
>     clip1=mp.VideoFileClip(path)
>     background = (mp.ColorClip((1280, 720), (0, 0, 0))
>         .set_duration(clip1.duration)
>         )
>     final=mp.CompositeVideoClip([background,clip1])
> ```

sorry but whats mp 
please share the full code or any reference

**Comment by shih6** [comment]
sorry 
import moviepy as mp
    clip1=mp.VideoFileClip(path)
    background = (mp.ColorClip((1280, 720), (0, 0, 0))
        .set_duration(clip1.duration)
        )
    final=mp.CompositeVideoClip([background,clip1])

**Comment by eswar159** [comment]
sorry but im getting errors
module 'moviepy' has no attribute 'VideoFileClip'
AttributeError: module 'moviepy' has no attribute 'ColorClip'
AttributeError: module 'moviepy' has no attribute 'CompositeVideoClip'

**Comment by shih6** [comment]
!!!!!

**Comment by shih6** [comment]
Wait a minute

**Comment by shih6** [comment]
https://zulko.github.io/moviepy/_modules/moviepy/video/VideoClip.html#ColorClip

**Comment by shih6** [comment]
my bad it was
import moviepy.editor as mp

**Comment by eswar159** [comment]
its ok buddy 
thank you will work on it!

**Comment by eswar159** [comment]
even after adding background, it's still enlarging the video

**Comment by shih6** [comment]
Any contact information？
do you have wechat?

**Comment by eswar159** [comment]
can u message me on whatsapp or facebook?
as i dont have wechat

**Comment by shih6** [comment]
whatsapp is also ok

**Comment by eswar159** [comment]
whatsapp :- +91 9441516627
Thank you

**Comment by YounesseAmhend** [comment]
you could just resized lol

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
